app/api/api_v1/routers/search.py

```diff
@@ -189,44 +188,51 @@ def download_all_search_documents(db=Depends(get_db)) -> StreamingResponse:
             detail="Missing required environment variables",
         )
 
-    aws_environment = "production" if "dev" not in PUBLIC_APP_URL else "staging"
-    data_dump_s3_key = f"navigator/{aws_environment}_data_dump_{INGEST_CYCLE_START}.csv"
+    data_dump_s3_key = "navigator/whole_data_dump.csv"
 
     s3_client = get_s3_client()
     valid_credentials = s3_client.is_connected()
+    if not valid_credentials:
+        _LOGGER.info("Error connecting to S3 AWS")
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED, detail="Error connecting to AWS"
+        )
 
     s3_document = S3Document(DOC_CACHE_BUCKET, AWS_REGION, data_dump_s3_key)
-    if not s3_client.document_exists(s3_document):
-        _LOGGER.info(f"Generating dump for ingest cycle w/c {INGEST_CYCLE_START}...")
-        df_as_csv = generate_data_dump_as_csv(db)
+    if valid_credentials is True and (not s3_client.document_exists(s3_document)):
+        aws_env = "production" if "dev" not in PUBLIC_APP_URL else "staging"
+        _LOGGER.info(
+            f"Generating {aws_env} dump for ingest cycle w/c {INGEST_CYCLE_START}..."
+        )
 
-        if valid_credentials is False:
-            _LOGGER.error("Cannot connect to AWS.")
-        else:
+        # After writing to a file buffer the position stays at the end whereas when you
+        # upload a buffer, it starts from the position it is currently in. We need to
+        # add the seek(0) to reset the buffer position to the beginning before writing
+        # to S3 to avoid creating an empty file.
+        df_as_csv = generate_data_dump_as_csv(INGEST_CYCLE_START, db)
+        df_as_csv.seek(0)
+
+        try:
             response = s3_client.upload_fileobj(
-                df_as_csv, DOC_CACHE_BUCKET, data_dump_s3_key
+                bucket=DOC_CACHE_BUCKET,
+                key=data_dump_s3_key,
+                content_type="application/csv",
+                fileobj=df_as_csv,
             )
             if response is False:
                 _LOGGER.error("Failed to upload object to s3: %s", response)
+        except Exception as e:
+            _LOGGER.error(e)
 
-        if s3_client.document_exists(s3_document):
-            _LOGGER.debug("Finished uploading data dump to s3")
-
-    else:
-        _LOGGER.debug("File already exists in S3. Fetching...")
-
-    s3_file = s3_client.download_file(s3_document)
-
-    _LOGGER.debug(f"Downloading all documents as of '{INGEST_CYCLE_START}' as CSV")
-    timestamp = datetime.now()
-    filename = f"whole_database_dump-{timestamp}.csv"
-    return StreamingResponse(
-        content=BytesIO(s3_file.read()),
-        headers={
-            "Content-Type": "text/csv",
-            "Content-Disposition": f"attachment; filename={filename}",
-        },
-    )
+    s3_document = S3Document(DOC_CACHE_BUCKET, AWS_REGION, data_dump_s3_key)
+    if s3_client.document_exists(s3_document):
```

Comment:
Really, really pedantic point - but prefer the "if" statements to be the exceptions to the expected flow. i.e. `if not s3_client.document_exists(s3_document): raise ... `